### PR TITLE
Translate adventure map objects

### PR DESCRIPTION
**Pull request by IvanSavenko**
Refactoring of object classes handler, to bring it in line with other handlers & to implement translation.

### CODE REVIEW

File: AI/VCAI/MapObjectsEvaluator.cpp [hunk 1]
```diff
@@ -32,10 +32,6 @@ MapObjectsEvaluator::MapObjectsEvaluator()
 				{
 					objectDatabase[CompoundMapObjectID(primaryID, secondaryID)] = handler->getAiValue().get();
 				}
-				else if(VLC->objtypeh->getObjGroupAiValue(primaryID) != boost::none) //if value is not initialized - fallback to default value for this object family if it exists
```
Comment (dydzio0614) [inline AI/VCAI/MapObjectsEvaluator.cpp:35]: why this got removed

Comment (IvanSavenko) [reply AI/VCAI/MapObjectsEvaluator.cpp:35]: AI value is now always in subtype

Comment (dydzio0614) [reply AI/VCAI/MapObjectsEvaluator.cpp:35]: but is there now a way to define default value for new subtypes added by mods?

Comment (IvanSavenko) [reply AI/VCAI/MapObjectsEvaluator.cpp:35]: Yes. Same way as for any other properties - "base" field.
See example in diff for `config/objects/moddables.json`

Comment (dydzio0614) [reply AI/VCAI/MapObjectsEvaluator.cpp:35]: is it something new? i thing this did not exist back in the day when I added ai value stuff

Comment (IvanSavenko) [reply AI/VCAI/MapObjectsEvaluator.cpp:35]: Nope. Always been here.

**Pull request merged**